tests/integration/clients/test_compass.py

```diff
@@ -0,0 +1,608 @@
+import time
+from random import choice
+from string import ascii_uppercase, hexdigits
+
+import pytest
+import requests
+
+from foundry_dev_tools.errors.compass import DuplicateNameError, ResourceNotTrashedError
+from foundry_dev_tools.errors.meta import FoundryAPIError
+from foundry_dev_tools.utils import api_types
+from tests.integration.conftest import TEST_SINGLETON
+from tests.integration.utils import INTEGRATION_TEST_COMPASS_ROOT_PATH, INTEGRATION_TEST_COMPASS_ROOT_RID
+
+
+def create_compass_folder() -> tuple[api_types.FolderRid, str]:
+    """Create a new folder and return its resource identifier."""
+
+    rnd = "".join(choice(ascii_uppercase) for i in range(5))
+    compass_folder_name = f"compass_folder_{rnd}"
+
+    response = TEST_SINGLETON.ctx.compass.api_create_folder(compass_folder_name, INTEGRATION_TEST_COMPASS_ROOT_RID)
+
+    assert response.status_code == 200
+
+    response_data = response.json()
+
+    rid = response_data["rid"]
+    name = response_data["name"]
+
+    return rid, name
+
+
+def delete_compass_folder(folder_rid: api_types.FolderRid) -> None:
+    """Tear-Down logic: Permanently Delete the compass folder with the associated `folder_rid`
+
+    Args:
+        folder_rid: The resource identifier of the folder to be deleted.
+
+    """
+
+    # Delete permanently with additional delete_options
+    response = TEST_SINGLETON.ctx.compass.api_delete_permanently(
+        {folder_rid}, delete_options={"DO_NOT_REQUIRE_TRASHED"}
+    )
+
+    assert response.status_code == 200
+
+
+@pytest.fixture()
+def compass_folder_setup_fixture():
+    # Create test folder
+    compass_folder_rid, compass_folder_name = create_compass_folder()
+
+    yield compass_folder_rid, compass_folder_name
+
+    # Delete test folder
+    delete_compass_folder(compass_folder_rid)
+
+
+@pytest.fixture()
+def marking_id():
+    """The id of `ls-use-case-foundry-devtools-dev-workspace` marking"""
+    return "3abebc75-4915-4b5c-8749-9ce06726b6e1"
+
+
+@pytest.fixture()
+def test_project_rid():
+    """The rid of `ls-use-case-foundry-devtools-dev-workspace` project"""
+    return "ri.compass.main.folder.f94fc087-6068-4e66-ac27-978fce7b9d9d"
```

Comment:
I think you could update INTEGRATION_TEST_COMPASS_ROOT_RID with a new folder in the ls-use-case-foundry-devtools-dev-workspaceand re-use that.

Reply:
Sure, I can reset the INTEGRATION_TEST_COMPASS_ROOT_RID to some new folder in the ls-use-case-foundry-devtools-dev-workspace project.
But perhaps it would also be good, to have the project rid defined as constant in `tests.integration.utils` as it is explicitly required in some tests, e.g. to fetch a project by its rid or adding/removing a marking on project level. It could also be achieved without the extra constant by extracting the project rid somehow from the information of the newly set INTEGRATION_TEST_COMPASS_ROOT_RID resource. But it would be some overhead. And in case people use their custom INTEGRATION_TEST_COMPASS_ROOT_RID it would also be a little bit more challenging as we may not know how to retrieve the rid of the `ls-use-case-foundry-devtools-dev-workspace` project.

Reply:
And maybe it is even better to not update the INTEGRATION_TEST_COMPASS_ROOT_RID with a folder in the ls-use-case-foundry-devtools-dev-workspace. The current INTEGRATION_TEST_COMPASS_ROOT_RID is more "accessible" compared to some folder rid in the ls-use-case-foundry-devtools-dev-workspace project. Not everyone has access to theproject and (whoever will work on the dev tools in future) might not be able to execute the tests then. 
But if you think, that it is not such a big deal, I can update the INTEGRATION_TEST_COMPASS_ROOT_RID of course.

Reply:
Fine for me to leave the INTEGRATION_TEST_COMPASS_ROOT_RID and add another constant pointing to the new folder.